src/ScraperBot/Crawler.php

```diff
@@ -102,5 +115,167 @@ public function crawlSites(SourceInterface $source, Client $client, $default_con
 
         $eachPromise->promise()->wait();
     }
-}
 
+    /**
+     * Return tag analysis of the given html as string.
+     *
+     * @param $body
+     * @return mixed
+     */
+    public function getTags($body){
+        if ($body != "") {
+            $dom = new \DOMDocument();
+            $dom->loadHTML($body, LIBXML_NOWARNING | LIBXML_NOERROR);
+            $allElements = $dom->getElementsByTagName('*');
+
+            $elementDistribution = [];
+            foreach($allElements as $element) {
+                if(array_key_exists($element->tagName, $elementDistribution)) {
+                    $elementDistribution[$element->tagName] += 1;
+                } else {
+                    $elementDistribution[$element->tagName] = 1;
+                }
+            }
+            $elementDistribution['total'] = $allElements->length;
+
+            return $elementDistribution;
+        }
+
+    }
+
+    /**
+     * Crawl sites.
+     *
+     * @param $listOfSites
+     * @param $client
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function crawlSiteMaps(SourceInterface $source, Client $client, $default_config = NULL, $timestamp = NULL, $offIndex = 0)
+    {
+        echo PHP_EOL . 'Sitemaps crawling>>>> ';
```

Comment:
If we start using echo, we're bypassing the symfony console.

Reply:
no, we should not. Thanks for raising it. I just need to check how to pass that class, maybe using DI?